v2/Dockerfile

```diff
@@ -1,4 +1,43 @@
-FROM quay.io/cybozu/ubuntu:20.04
+FROM --platform=$BUILDPLATFORM golang:1.18-buster as build-env
+
+ARG TARGETARCH
+
+COPY . /workdir
+WORKDIR /workdir
+
+RUN GOARCH=${TARGETARCH} CGO_ENABLED=0 go build -o work/coil -ldflags="-s -w" cmd/coil/*.go
```

Comment:
Is there a specific requirement for executing every build step in its own `RUN` statement? As it is you'll end up with a bunch of extraneous (albeit harmless as this is the build stage) layers.

(suggestion) On a related note, since you are copying over the entire project directory into this build stage, perhaps it would be better to have all the `go build` commands placed into the `make build` Makefile target. This would make it easier for developpers to build `coil` locally.